packages/grafana-data/src/types/config.ts

```diff
@@ -85,6 +86,45 @@ export type OAuth =
  */
 export type OAuthSettings = Partial<Record<OAuth, { name: string }>>;
 
+/** Current user info included in bootData
+ *
+ * @public
```

Comment:
@mckn @torkelo -- what are your thoughs on this as a "Public" API -- I guess it is accidentally

The backend API is periodically adding/changing things with this kind of object -- that I don't really want to be commitd to as a public API

Any thoughts on how to best communicate that you can count on:
* login
* email
* name
But that
* helpFlags1
* hasEditPermissionsInFolder 
* etc
are not to be counted on for long term

Reply:
Do we really want to expose all these information to the plugins?

I would probably say that we would like to provide more specific APIs to access these (or a subset of them) from plugins. Of course it's a bit hard to make it truly internal until it lives in `@grafana/data`.

On the other hand as it is indirectly exposed through the `GrafanaConfig`, does `@public` make any difference here?

Reply:
Yeah that's fair, maybe we could expose a subset of the interface, and then the "full" interface extending it is kept private? Or we could use jsdoc annotations to specify these properties shouldn't be used or depended on?

Reply:
Yeah, it is part of the `BootData` and hence it is part of that API and should have the same release tag as that.

Maybe we can do something in lines with what @kaydelaney suggests?

Reply:
Hm.. not sure we should encurage usage of config.bootData , it's a hack/smell

not sure this should be ever used directly, boot data is only to setup the config object.

Boot data was only intended to bootstrap the config object and contextSrv , so using it directly after feels a bit hacky.

So accessing config.bootData.user feels wrong, either we should add user to config object or make something like contextSrv accessible to plugins. Maybe userSrv? it looks like all uses of config.bootData is to access current user (Which should be done using contextSrv at least historically). But I am open to suggestions for how to expose current user. What is best?

Adding typing to bootData is useful but not sure how we limit it's use / exposure externally

Reply:
This is not _currently_ a public API is it? 

Can we just keep this as private/internal?

The data itself has always been just "publicly available" on window, right, so there's no change in actual exposure.